### PR TITLE
 #2570 Apply group concept for broker

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2570)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-agent/src/main/scala/com/island/ohara/agent/BrokerCollie.scala [hunk 1]
```diff
@@ -382,9 +379,9 @@ object BrokerCollie {
 
     override def create(): Future[BrokerClusterInfo] = {
       // initial the basic creation required parameters (defined in ClusterInfo) for broker
-      //TODO : it is ok that we don't add the group() since client will auto fill default value for us
-      //TODO: add group() in #2570 for broker
-      val creation = request.name(clusterName).imageName(imageName).nodeNames(nodeNames).creation
+      // Note: We add all the fields in collie and they may not required in client api
```
Comment (chia7712) [inline ohara-agent/src/main/scala/com/island/ohara/agent/BrokerCollie.scala]: nice comment!

**Comment by saivirtue** [comment]
> seems we are able to make collie.creator depend on Api.request. If so, the fields in basic creator can be removed and all setter can be abstract. The sub classes should have their way to set nodeNames, group, name and so on. The benefit is that the creator get pure interface and we can get rid of weird helper method (for example, doCopy).

This needs a sub-task to finish. Already added in #2570

**Comment by saivirtue** [comment]
retry it

**Comment by saivirtue** [comment]
retry it

**Pull request merged**